packages/server/lib/controllers/config.controller.ts

```diff
@@ -355,22 +366,38 @@ class ConfigController {
                 const [availableFlowsForProvider] = availableFlows.filter((flow) => flow.providerConfigKey === config.provider);
 
-                const enabledFlows = await getConfigWithEndpointsByProviderConfigKey(environmentId, providerConfigKey);
-                const unEnabledFlows: StandardNangoConfig = availableFlowsForProvider as StandardNangoConfig;
-
-                if (availableFlows && enabledFlows && unEnabledFlows) {
-                    const { syncs: enabledSyncs, actions: enabledActions } = enabledFlows;
-
-                    const { syncs, actions } = unEnabledFlows;
-
-                    const filteredSyncs = syncs.filter((sync) => !enabledSyncs.some((enabledSync) => enabledSync.name === sync.name));
-                    const filteredActions = actions.filter((action) => !enabledActions.some((enabledAction) => enabledAction.name === action.name));
-
-                    unEnabledFlows.syncs = filteredSyncs;
-                    unEnabledFlows.actions = filteredActions;
+                const enabledAndDisabledFlows = await getConfigWithEndpointsByProviderConfigKey(environmentId, providerConfigKey);
+                const publicFlows: StandardNangoConfig = availableFlowsForProvider as StandardNangoConfig;
+                const disabledFlows = publicFlows;
+
+                if (availableFlows && enabledAndDisabledFlows && publicFlows) {
+                    const { syncs, actions } = enabledAndDisabledFlows;
+                    const { enabledSyncs, disabledSyncs } = syncs.reduce<SyncConfigs>(
+                        (acc, sync: NangoSyncConfig) => (sync.enabled ? acc.enabledSyncs.push(sync) : acc.disabledSyncs.push(sync), acc),
+                        { enabledSyncs: [], disabledSyncs: [] }
+                    );
+                    const { enabledActions, disabledActions } = actions.reduce<ActionConfigs>(
+                        (acc, action: NangoSyncConfig) => (action.enabled ? acc.enabledActions.push(action) : acc.disabledActions.push(action), acc),
+                        { enabledActions: [], disabledActions: [] }
+                    );
+
+                    enabledAndDisabledFlows.syncs = enabledSyncs;
+                    enabledAndDisabledFlows.actions = enabledActions;
+
+                    const { syncs: publicSyncs, actions: publicActions } = publicFlows;
+
+                    const filteredSyncs = publicSyncs.filter(
+                        (sync) => ![...enabledSyncs, ...disabledSyncs].some((enabledSync) => enabledSync.name === sync.name)
+                    );
+                    const filteredActions = publicActions.filter(
+                        (action) => ![...enabledActions, ...disabledActions].some((enabledAction) => enabledAction.name === action.name)
+                    );
+
+                    disabledFlows.syncs = [...filteredSyncs, ...disabledSyncs];
+                    disabledFlows.actions = [...filteredActions, ...disabledActions];
```

Comment:
exctracting this block into its own function with maybe a comment might make it easier to understand. Maybe it is just me but I had to look at it for a long time to fully grasp what's going on here

Reply:
Done with https://github.com/NangoHQ/nango/pull/1993/commits/3dbb6110054b7df5b935a995cf17b5764773c1f8